src/Generator/Aggregator.fs

```diff
@@ -0,0 +1,83 @@
+module Aggregator
+
+open Types
+open Util
+open SpecificationOpenGL
+open System
+
+// Generically take the field name
+let inline nameField a = (^a: (member Name: string) (a))
+
+let getEnumCasesAndCommandsPerVersion (data: OpenGL_Specification.Registry) =
+    let featuresSortedAscendingByVersion =
+        data.Features |> Array.sortBy (fun data -> data.Number)
+    let versions =
+        featuresSortedAscendingByVersion
+        |> Array.Parallel.map (fun data -> data.Name)
+    let resultsByVersion = Array.zeroCreate versions.Length
+    let aggregatedResultByVersion = Array.zeroCreate versions.Length
+    let getVersionBefore index = aggregatedResultByVersion.[index - 1]
+
+    let extractAddedAndRemoved (data: OpenGL_Specification.Feature) =
+        let inline divideIntoCommandsAndEnums cmd enum e =
+            [| Left(cmd e |> Array.Parallel.map nameField)
+               Right(enum e |> Array.Parallel.map nameField) |]
+
+        let (addedFunctions: string [], addedEnums: string []) =
+            let a, b =
```

Comment:
the a,b tuple generative function should be abstracted in a single function which is called twice to keep code dry.

Reply:
I guess the function I'm searching for is `||>`

Reply:
That'd be good! (a,b) ||> functionToCreateTuple
Then no need to recreate a,b again afterwards:)